minimint/src/net/api.rs

```diff
@@ -105,13 +105,13 @@ fn server_endpoints() -> &'static [ApiEndpoint<MinimintConsensus<rand::rngs::OsR
             async |minimint: &MinimintConsensus<rand::rngs::OsRng>, transaction: serde_json::Value| -> TransactionId {
                 // deserializing Transaction from json Value always fails
                 // we need to convert it to string first
-                let string = serde_json::to_string(&transaction).expect("encoding error");
+                let string = serde_json::to_string(&transaction).map_err(|e| ApiError::bad_request(e.to_string()))?;
```

Comment:
This should never happen. Because `serde_json::Value` doesn't values that are not convertable to a json string.